src/rules/store/select-in-selector-names.ts

```diff
@@ -0,0 +1,76 @@
+import path from 'path'
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils'
+import { isIdentifier, isLiteral } from '../../utils/helper-functions/index'
+
+import { memorizedSelector, selectorsNames, docsUrl } from '../../utils'
+
+export const messageId = 'selectInSelectorNames'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: path.parse(__filename).name,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      category: 'Best Practices',
+      description: 'Name Selector Functions as selectThing',
+      recommended: 'warn',
+    },
+    schema: [],
+    messages: {
+      [messageId]: 'Name Selector Functions as selectThing',
+    },
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const includedKeyword = 'select'
+
+    function hasKeyword(name: string): boolean {
+      // console.log(name.toLowerCase().startsWith(includedKeyword))
+      return name.toLowerCase().startsWith(includedKeyword)
+    }
+
+    return {
+      [`VariableDeclaration[declarations.length>0]`](
+        node: TSESTree.VariableDeclaration,
+      ) {
+        for (const dec of node.declarations) {
+          const id: TSESTree.Identifier = (dec.id as any) as TSESTree.Identifier
+          // console.log(id.name) //, dec.id.typeAnnotation?.typeAnnotation)
+          if (dec.id.typeAnnotation?.typeAnnotation) {
+            if (
+              /^${memorizedSelectorTypes}$/.exec(
```

Comment:
Instead of `exec`, it's better to use `test` this will return a `true` or a `false` and not the match.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

Reply:
To complement this comment from @timdeschryver, it worths to say that `/^${memorizedSelectorTypes}$/` and `/^${selectorsNames}$/` don't work the way you might be imagining... if you do it like this, you'll get the words `memorizedSelectorTypes` and `selectorsNames` instead of the computed variables.

When you want to interpolate a variable into a `RegExp` you have to do this instead:

```ts
RegExp(`^${memorizedSelectorTypes}$`)
```

and 

```ts
RegExp(\`^${selectorsNames}$`)
```